ci/infra/testrunner/tests/conftest.py

```diff
@@ -12,6 +23,97 @@ def pytest_addoption(parser):
     parser.addoption("--vars", action="store",help="vars yaml" )
     parser.addoption("--platform", action="store",help="target platform" )
 
+
+class Workspace:
+    def __init__(self):
+        self.temporary_directory = tempfile.mkdtemp(prefix="skuba-tests")
+
+    def cleanup(self):
+        shutil.rmtree(self.temporary_directory)
+
+
+class ClusterFixture(contextlib.AbstractContextManager):
+    """Wrap cluster creation and bootstrapping into a context manager with state.
+
+    This allows creating a cluster using this fixture that will keep track of it's
+    own workspace:
+
+    >>> with ClusterFixture(masters=1, workers=1, platform="openstack") as c:
+    >>>     c.run("kubectl ...")
+
+    Once 'c' goes out of scope the cluster directory from skuba will be removed and
+    the terraform deployment will be destroyed.
+    """
+    def __init__(self, conf: BaseConfig, masters=1, workers=1, platform='openstack'):
+        self.masters = masters
+        self.workers = workers
+        self.workspace = Workspace()
+        self.config = BaseConfig(conf.yaml_path)
```

Comment:
I Think we are breaking a lot the encapsulation of concerns by instantiating the BaseConfig class here. Not sure if this is necessary. The configuration is available by means of the `conf` fixture.

Reply:
Well the current testrunner setup breaks, when attempting to run tests in parallel (in Jenkins), because they all attempt to use the same workspace - that's why this is make a shallow copy of the configuration and after that resets the configuration paths for the workspace to point to a temporary directory.

If I just reset the directories on the initial `conf` reference, obviously all usages would get the same temporary directory and it would be useless.